include/libopencm3/efm32/hg/cmu.h

```diff
@@ -0,0 +1,674 @@
+#define CMU_HFCORECLKDIV_HFCORECLKDIV_HFCLK512	\
+	CMU_HFCORECLKDIV_HFCORECLKDIV(9)
+
+#define CMU_HFCORECLKDIV_HFCORECLKDIV_NODIV	\
+	CMU_HFCORECLKDIV_HFCORECLKDIV_HFCLK
+#define CMU_HFCORECLKDIV_HFCORECLKDIV_DIV2	\
+	CMU_HFCORECLKDIV_HFCORECLKDIV_HFCLK2
+#define CMU_HFCORECLKDIV_HFCORECLKDIV_DIV4	\
+	CMU_HFCORECLKDIV_HFCORECLKDIV_HFCLK4
+#define CMU_HFCORECLKDIV_HFCORECLKDIV_DIV8	\
+	CMU_HFCORECLKDIV_HFCORECLKDIV_HFCLK8
+#define CMU_HFCORECLKDIV_HFCORECLKDIV_DIV16	\
+	CMU_HFCORECLKDIV_HFCORECLKDIV_HFCLK16
+#define CMU_HFCORECLKDIV_HFCORECLKDIV_DIV32	\
+	CMU_HFCORECLKDIV_HFCORECLKDIV_HFCLK32
+#define CMU_HFCORECLKDIV_HFCORECLKDIV_DIV64	\
+	CMU_HFCORECLKDIV_HFCORECLKDIV_HFCLK64
+#define CMU_HFCORECLKDIV_HFCORECLKDIV_DIV128	\
+	CMU_HFCORECLKDIV_HFCORECLKDIV_HFCLK128
+#define CMU_HFCORECLKDIV_HFCORECLKDIV_DIV256	\
+	CMU_HFCORECLKDIV_HFCORECLKDIV_HFCLK256
+#define CMU_HFCORECLKDIV_HFCORECLKDIV_DIV512	\
+	CMU_HFCORECLKDIV_HFCORECLKDIV_HFCLK512
+
+/* CMU_HFPERCLKDIV */
+#define CMU_HFPERCLKDIV_HFPERCLKEN		(1 << 8)
+
+#define CMU_HFPERCLKDIV_HFPERCLKDIV_SHIFT	(0)
+#define CMU_HFPERCLKDIV_HFPERCLKDIV_MASK	\
+	(0xF << CMU_HFPERCLKDIV_HFPERCLKDIV_SHIFT)
+#define CMU_HFPERCLKDIV_HFPERCLKDIV(v)		\
+	(((v) << CMU_HFPERCLKDIV_HFPERCLKDIV_SHIFT) & \
+	 CMU_HFPERCLKDIV_HFPERCLKDIV_MASK)
+#define CMU_HFPERCLKDIV_HFPERCLKHFCLK_HFCLK	CMU_HFPERCLKDIV_HFPERCLKDIV(0)
+#define CMU_HFPERCLKDIV_HFPERCLKHFCLK_HFCLK2	CMU_HFPERCLKDIV_HFPERCLKDIV(1)
+#define CMU_HFPERCLKDIV_HFPERCLKHFCLK_HFCLK4	CMU_HFPERCLKDIV_HFPERCLKDIV(2)
+#define CMU_HFPERCLKDIV_HFPERCLKHFCLK_HFCLK8	CMU_HFPERCLKDIV_HFPERCLKDIV(3)
+#define CMU_HFPERCLKDIV_HFPERCLKHFCLK_HFCLK16	CMU_HFPERCLKDIV_HFPERCLKDIV(4)
+#define CMU_HFPERCLKDIV_HFPERCLKHFCLK_HFCLK32	CMU_HFPERCLKDIV_HFPERCLKDIV(5)
+#define CMU_HFPERCLKDIV_HFPERCLKHFCLK_HFCLK64	CMU_HFPERCLKDIV_HFPERCLKDIV(6)
+#define CMU_HFPERCLKDIV_HFPERCLKHFCLK_HFCLK128	CMU_HFPERCLKDIV_HFPERCLKDIV(7)
+#define CMU_HFPERCLKDIV_HFPERCLKHFCLK_HFCLK256	CMU_HFPERCLKDIV_HFPERCLKDIV(8)
+#define CMU_HFPERCLKDIV_HFPERCLKHFCLK_HFCLK512	CMU_HFPERCLKDIV_HFPERCLKDIV(9)
+
+/* CMU_HFPERCLKDIV_HFPERCLKHFCLK_HFCLK* to CMU_HFPERCLKDIV_HFPERCLKHFCLK_DIV* */
+#define CMU_HFPERCLKDIV_HFPERCLKDIV_NODIV	\
+	CMU_HFPERCLKDIV_HFPERCLKDIV_HFCLK
+#define CMU_HFPERCLKDIV_HFPERCLKDIV_DIV2	\
+	CMU_HFPERCLKDIV_HFPERCLKDIV_HFCLK2
+#define CMU_HFPERCLKDIV_HFPERCLKDIV_DIV4	\
+	CMU_HFPERCLKDIV_HFPERCLKDIV_HFCLK4
+#define CMU_HFPERCLKDIV_HFPERCLKDIV_DIV8	\
+	CMU_HFPERCLKDIV_HFPERCLKDIV_HFCLK8
+#define CMU_HFPERCLKDIV_HFPERCLKDIV_DIV16	\
+	CMU_HFPERCLKDIV_HFPERCLKDIV_HFCLK16
+#define CMU_HFPERCLKDIV_HFPERCLKDIV_DIV32	\
+	CMU_HFPERCLKDIV_HFPERCLKDIV_HFCLK32
+#define CMU_HFPERCLKDIV_HFPERCLKDIV_DIV64	\
+	CMU_HFPERCLKDIV_HFPERCLKDIV_HFCLK64
+#define CMU_HFPERCLKDIV_HFPERCLKDIV_DIV128	\
+	CMU_HFPERCLKDIV_HFPERCLKDIV_HFCLK128
+#define CMU_HFPERCLKDIV_HFPERCLKDIV_DIV256	\
+	CMU_HFPERCLKDIV_HFPERCLKDIV_HFCLK256
+#define CMU_HFPERCLKDIV_HFPERCLKDIV_DIV512	\
+	CMU_HFPERCLKDIV_HFPERCLKDIV_HFCLK512
```

Comment:
what on earth is this garbage?!  I mean, ok, it will get grandfathered, but yuck!